packages/nodes-base/nodes/Google/Gmail/Gmail.node.ts

```diff
@@ -0,0 +1,386 @@
+	description: INodeTypeDescription = {
+		displayName: 'Gmail',
+		name: 'gmail',
+		icon: 'file:gmail.png',
+		group: ['transform'],
+		version: 1,
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		description: 'Consume the Gmail API',
+		defaults: {
+			name: 'Gmail',
+			color: '#d93025',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'gmailOAuth2',
+				required: true,
+			},
+		],
+		properties: [
+			{
+				displayName: 'Resource',
+				name: 'resource',
+				type: 'options',
+				options: [
+					{
+						name: 'Draft',
+						value: 'draft',
+					},
+					{
+						name: 'Label',
+						value: 'label',
+					},
+					{
+						name: 'Message',
+						value: 'message',
+					},
+				],
+				default: 'draft',
+				description: 'The resource to operate on.',
+			},
+			//-------------------------------
+			// Draft Operations
+			//-------------------------------
+			...draftOperations,
+			...draftFields,
+			//-------------------------------
+			// Label Operations
+			//-------------------------------
+			...labelOperations,
+			...labelFields,
+			//-------------------------------
+			// Message Operations
+			//-------------------------------
+			...messageOperations,
+			...messageFields,
+		]
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		const resource = this.getNodeParameter('resource', 0) as string;
+		const operation = this.getNodeParameter('operation', 0) as string;
+
+		let method = '';
+		let body: IDataObject = {};
+		let qs: IDataObject = {};
+		let endpoint = '';
+		let responseData;
+
+		for (let i = 0; i < items.length; i++) {
+			//-------------------------------
+			// Label Operations
+			//-------------------------------
+			if(resource === 'label') {
+				//-------------------------------
+				// Create Label
+				//-------------------------------
+				if(operation === 'create') {
+					// https://developers.google.com/gmail/api/v1/reference/users/labels/create
+					const labelName = this.getNodeParameter('name', i) as string;
+
+					method = 'POST';
+					endpoint = '/gmail/v1/users/me/labels';
+
+					body = {
+						labelListVisibility: 'labelShow',
+						messageListVisibility: 'show',
+						name: labelName,
+					};
+
+					responseData = await googleApiRequest.call(this, method, endpoint, body, qs);
+				//-------------------------------
+				// Add/Remove Label
+				//-------------------------------
+				} else if(['add', 'delete'].includes(operation)) {
+					// https://developers.google.com/gmail/api/v1/reference/users/messages/modify
+					const messageID = this.getNodeParameter('messageId', i);
+					const labelID = this.getNodeParameter('labelId', i);
+
+					method = 'POST';
+					endpoint = `/gmail/v1/users/me/messages/${messageID}/modify`;
+
+					if(operation === 'add') {
+						body = {
+							'addLabelIds': [
```

Comment:
Super minor detail, but no need for quotes in this kind of properties. Same for below.